tests/foreman/api/test_smartproxy.py

```diff
@@ -14,14 +14,224 @@
+    default_url_on_new_port,
+    get_available_capsule_port,
 )
 from robottelo.test import APITestCase
-from robottelo.api.utils import one_to_many_names
+
+
+class CapsuleTestCase(APITestCase):
+    """Tests for Smart Proxy (Capsule) entity."""
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_url(self):
+        """Proxy creation with random URL
+
+        @id: e48a6260-97e0-4234-a69c-77bbbcde85d6
+
+        @Assert: Proxy is not created
+        """
+        # Create a random proxy
+        with self.assertRaises(HTTPError) as context:
+            entities.SmartProxy(
+                url='http://{0}:{1}'.format(
+                    gen_string('alpha', 6),
+                    gen_string('numeric', 4)
+                )).create()
+        self.assertRegexpMatches(
+            context.exception.response.text, u'Unable to communicate')
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_name(self):
+        """Proxy creation with valid name
+
+        @id: 0ffe0dc5-675e-45f4-b7e1-a14d3dd81f6e
+
+        @Assert: Proxy is created
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                with default_url_on_new_port(
+                        9090, get_available_capsule_port()) as url:
+                    proxy = entities.SmartProxy(name=name, url=url).create()
+                    self.assertEquals(proxy.read().name, name)
+                # Add capsule id to cleanup list
+                self.addCleanup(capsule_cleanup, proxy.id)
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_positive_delete(self):
+        """Proxy deletion
+
+        @id: 872bf12e-736d-43d1-87cf-2923966b59d0
+
+        @Assert: Proxy is deleted
+        """
+        with default_url_on_new_port(
+                9090, get_available_capsule_port()) as url:
+            proxy = entities.SmartProxy(url=url).create()
+        if bz_bug_is_open(1398695):
+            try:
+                proxy.delete()
+            except HTTPError as err:
+                if "Can't modify frozen hash" not in err.response.text:
+                    raise err
+        else:
+            proxy.delete()
+        with self.assertRaises(HTTPError):
+            proxy.read()
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_name(self):
+        """Proxy name update
+
+        @id: f279640e-d7e9-48a3-aed8-7bf406e9d6f2
+
+        @Assert: Proxy has the name updated
+        """
+        with default_url_on_new_port(
+                9090, get_available_capsule_port()) as url:
+            proxy = entities.SmartProxy(url=url).create()
+            for new_name in valid_data_list():
+                with self.subTest(new_name):
+                    proxy.name = new_name
+                    proxy.update(['name'])
```

Comment:
`update()` logic performs `read()` call as well, but, probably, it is not so big problem to perform two read() in a row. Just a nitpick